### PR TITLE
ci(github): add github action ci

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

**Comment by douglasduteil** [comment]
@yohanboniface Is it compatible with CPython (3.9.0) ?

**Comment by douglasduteil** [comment]
@yohanboniface @magopian What is missing to make it work ?

**Comment by yohanboniface** [comment]
This commit https://github.com/SocialGouv/egapro-api/commit/9a98b3b4c560088b813b071bed4fc750fe66b1fb

(Which is on schema branch, which is the active branch, and next release)

**Comment by douglasduteil** [comment]
@yohanboniface rebase on `schema` done

**Comment by douglasduteil** [comment]
The tests are not running ...

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: CI
+on:
+  - push
+  - pull_request
+
+jobs:
+  Quality:
+    strategy:
+      # Don't fast-fail on tag and master
```
Comment (yohanboniface) [inline .github/workflows/ci.yaml:9]: Can you explain (briefly) the needs behind this ?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          python-version: '3.8'
+
+      - name: Save pip cache dir
+        id: init
+        run: echo "::set-output name=pip_cache::$(pip cache dir)"
+        
+      - uses: actions/checkout@v2
+      
+      - name: Cache Pip packages
+        id: yarn_cache_packages
```
Comment (yohanboniface) [inline .github/workflows/ci.yaml:43]: yarn ? :)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        ports:
+          - 5432:5432
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Save pip cache dir
```
Comment (yohanboniface) [inline .github/workflows/ci.yaml:36]: I don't get this step :s

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:11-alpine
+        env:
+          POSTGRES_DB: test_egapro
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        options: >-
```
Comment (yohanboniface) [inline .github/workflows/ci.yaml:22]: Do we really need those options ? The less config, the better :)

**Review by yohanboniface — COMMENTED** [review]
Thanks for the work!

I've added a bit of comments/remarks :)

**Comment by yohanboniface** [comment]
> The tests are not running ...

Ah, my bad! I've renamed "tests" dir to "test", but so we show either make the command phony or rename it the other way around.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (douglasduteil) [reply .github/workflows/ci.yaml]: Might not be needed actually

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (douglasduteil) [reply .github/workflows/ci.yaml]: This step will output un `pip_cache` `init` step id `outputs` 
Used as `${{ steps.init.outputs.pip_cache }}`in https://github.com/SocialGouv/egapro-api/pull/17/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR35

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (douglasduteil) [reply .github/workflows/ci.yaml]: Removed

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (douglasduteil) [reply .github/workflows/ci.yaml]: removed

**Comment by yohanboniface** [comment]
### TODO:

- [ ] coverage not generated
- [ ] make that we do not run twice the CI (once for PR, once for push)

**Comment by douglasduteil** [comment]
>     * make that we do not run twice the CI (once for PR, once for push)

Is accessory...
The double check is normal as it's testing two different scenarios : the branch itself and the merge commit on the target branch. 
If your are really really really brother by it and you can't rest knowing that two similar checks are made per commits on local branch prs. Just remove the line 4 `- pull_request` 👍

**Comment by yohanboniface** [comment]
> the merge commit on the target branch.

Humm, are we sure about that ?

**Comment by douglasduteil** [comment]
[Yep.](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows) The goal is to be able ton test forks but I didn't tested it.

**Comment by douglasduteil** [comment]
🎉

**Pull request merged**